src/api/spec/models/project_spec.rb

```diff
@@ -313,4 +313,99 @@
       expect(Project.deleted?('never-existed-before')).to be_truthy
     end
   end
+
+  describe '.restore' do
+    let!(:admin_user) { create(:admin_user, login: 'Admin') }
+    let!(:project_name) { 'project_used_for_restoration' }
+    let(:project_meta) {
+      <<-PROJECT_META_XML
+        <project name=\"#{project_name}\">\n  <title></title>\n
+        <description></description>\n  <person userid=\"Admin\" role=\"maintainer\" />\n</project>
+      PROJECT_META_XML
+    }
+
+    before do
+      login admin_user
+
+      allow(Backend::Connection).to receive(:post)
+      allow_any_instance_of(ProjectMetaFile).to receive(:to_s).and_return(project_meta)
```

Comment:
IMO this depends to much on implementation details you just happen to know. Could we project before and then test it?